controllers/openstackmachine_controller.go

```diff
@@ -353,7 +354,23 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, logger
 			handleUpdateMachineError(logger, openStackMachine, errors.Errorf("Floating IP cannot be got or created: %v", err))
 			return ctrl.Result{}, nil
 		}
-		err = computeService.AssociateFloatingIP(instance.ID, fp.FloatingIP)
+
+		// There is some possibility that floating IP is created but not able to be find right after that
+		// so give a retry here, see #927 for further information.
+		backoff := wait.Backoff{
+			Steps:    10,
+			Duration: time.Second,
+			Factor:   1.25,
+			Jitter:   0.1,
+		}
+		err = wait.ExponentialBackoff(backoff, func() (bool, error) {
+			if err := computeService.AssociateFloatingIP(instance.ID, fp.FloatingIP); err != nil {
+				logger.Info("Floating IP association failed, will retry.", "instance-id", instance.ID, "floating-ip", fp.FloatingIP)
+				return false, err
```

Comment:
returning `false, err` will lead to the backoff to stop instead of doing the next retry.
So we have to log only and do `return false, nil`. The inner error could be set to a error variable defined outside the retryFunc so it could get added as infromation in line 375 if wanted.